pkg/profiler/profiler.go

```diff
@@ -712,22 +713,24 @@ func (p *CgroupProfiler) sendProfile(ctx context.Context, prof *profile.Profile)
 		return err
 	}
 
-	var labeloldformat []*profilestorepb.Label
-
+	var (
+		labelOldFormat = make([]*profilestorepb.Label, 0, len(p.Labels()))
+		i              = 0
+	)
 	for key, value := range p.Labels() {
-		labeloldformat = append(labeloldformat,
-			&profilestorepb.Label{
-				Name:  string(key),
-				Value: string(value),
-			})
+		labelOldFormat[i] = &profilestorepb.Label{
```

Comment:
Since we have a slice with capacity=`len(p.Labels())`
 and length 0, we need to use `append` to avoid writing out of bounds, no? :D 

> 

https://go.dev/play/p/17_G4OQD4eU